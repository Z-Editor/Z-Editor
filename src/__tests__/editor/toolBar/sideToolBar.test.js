import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SideToolBar from '../../../editor/toolBar/sideToolBar';
import { l_config } from '../../../__mocks__/config.mock';

configure({ adapter: new Adapter() });

describe('renders without crashing', () => {
  const mockInsertFn = jest.fn();

  const sideToolBar = shallow(<SideToolBar side={'left'} data={l_config} insertFn={mockInsertFn} />);

  it('renders correctly', () => {
    expect(sideToolBar).toMatchSnapshot();
    expect(sideToolBar.find('ul').length).toBe(1);
    expect(sideToolBar.find('ul').find('SideToolBarBtn').length).toBe(l_config.length);

    for (let i = 0; i < l_config.length; i++) {
      expect(
        sideToolBar
          .find('ul')
          .find('SideToolBarBtn')
          .get(i).props.side
      ).toEqual('left');
      expect(
        sideToolBar
          .find('ul')
          .find('SideToolBarBtn')
          .get(i).props.type
      ).toEqual(l_config[i]);
    }
  });
});
