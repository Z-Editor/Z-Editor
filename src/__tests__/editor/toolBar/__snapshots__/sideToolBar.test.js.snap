// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideToolBar
    data={
      Array [
        Object {
          "icon": "full.svg",
          "id": 100,
          "type": "main",
        },
        Object {
          "icon": "half.svg",
          "id": 101,
          "type": "half",
        },
        Object {
          "icon": "bar.svg",
          "id": 102,
          "type": "bar",
        },
        Object {
          "icon": "inverse.svg",
          "id": 103,
          "type": "inverse",
        },
      ]
    }
    insertFn={[MockFunction]}
    side="left"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SideToolBarBtn
          insertFn={[MockFunction]}
          side="left"
          type={
            Object {
              "icon": "full.svg",
              "id": 100,
              "type": "main",
            }
          }
        />,
        <SideToolBarBtn
          insertFn={[MockFunction]}
          side="left"
          type={
            Object {
              "icon": "half.svg",
              "id": 101,
              "type": "half",
            }
          }
        />,
        <SideToolBarBtn
          insertFn={[MockFunction]}
          side="left"
          type={
            Object {
              "icon": "bar.svg",
              "id": 102,
              "type": "bar",
            }
          }
        />,
        <SideToolBarBtn
          insertFn={[MockFunction]}
          side="left"
          type={
            Object {
              "icon": "inverse.svg",
              "id": 103,
              "type": "inverse",
            }
          }
        />,
      ],
      "className": "rdw-editor-toolbar-side",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "100",
        "nodeType": "function",
        "props": Object {
          "insertFn": [MockFunction],
          "side": "left",
          "type": Object {
            "icon": "full.svg",
            "id": 100,
            "type": "main",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "101",
        "nodeType": "function",
        "props": Object {
          "insertFn": [MockFunction],
          "side": "left",
          "type": Object {
            "icon": "half.svg",
            "id": 101,
            "type": "half",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "102",
        "nodeType": "function",
        "props": Object {
          "insertFn": [MockFunction],
          "side": "left",
          "type": Object {
            "icon": "bar.svg",
            "id": 102,
            "type": "bar",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "103",
        "nodeType": "function",
        "props": Object {
          "insertFn": [MockFunction],
          "side": "left",
          "type": Object {
            "icon": "inverse.svg",
            "id": 103,
            "type": "inverse",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SideToolBarBtn
            insertFn={[MockFunction]}
            side="left"
            type={
              Object {
                "icon": "full.svg",
                "id": 100,
                "type": "main",
              }
            }
          />,
          <SideToolBarBtn
            insertFn={[MockFunction]}
            side="left"
            type={
              Object {
                "icon": "half.svg",
                "id": 101,
                "type": "half",
              }
            }
          />,
          <SideToolBarBtn
            insertFn={[MockFunction]}
            side="left"
            type={
              Object {
                "icon": "bar.svg",
                "id": 102,
                "type": "bar",
              }
            }
          />,
          <SideToolBarBtn
            insertFn={[MockFunction]}
            side="left"
            type={
              Object {
                "icon": "inverse.svg",
                "id": 103,
                "type": "inverse",
              }
            }
          />,
        ],
        "className": "rdw-editor-toolbar-side",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "100",
          "nodeType": "function",
          "props": Object {
            "insertFn": [MockFunction],
            "side": "left",
            "type": Object {
              "icon": "full.svg",
              "id": 100,
              "type": "main",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "101",
          "nodeType": "function",
          "props": Object {
            "insertFn": [MockFunction],
            "side": "left",
            "type": Object {
              "icon": "half.svg",
              "id": 101,
              "type": "half",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "102",
          "nodeType": "function",
          "props": Object {
            "insertFn": [MockFunction],
            "side": "left",
            "type": Object {
              "icon": "bar.svg",
              "id": 102,
              "type": "bar",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "103",
          "nodeType": "function",
          "props": Object {
            "insertFn": [MockFunction],
            "side": "left",
            "type": Object {
              "icon": "inverse.svg",
              "id": 103,
              "type": "inverse",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
